File: mobile/app/src/main/java/com/dinder/app/domain/model/Notification.kt
package com.dinder.app.domain.model

/**
 * Domain model representing an in-app notification.
 */
data class Notification(
    val notificationId: String,
    val type: String, // "Match" | "Message" | "Promotion"
    val title: String,
    val body: String? = null,
    val deepLinkPayload: String? = null,
    val isRead: Boolean = false,
    val createdAt: String
)

File: old/src/Dinder.Domain/Entities/Notification.cs
using Dinder.Domain.Enums;

namespace Dinder.Domain.Entities;

public sealed class Notification
{
    public Guid Id { get; private set; }
    public Guid UserId { get; private set; }
    public NotificationType Type { get; private set; }
    public string Title { get; private set; }
    public string? Body { get; private set; }
    public string? DeepLinkPayload { get; private set; }
    public bool IsRead { get; private set; }
    public DateTime CreatedAt { get; private set; }

#pragma warning disable CS8618
    private Notification() { } // EF Core
#pragma warning restore CS8618

    public Notification(Guid userId, NotificationType type, string title, string? body = null, string? deepLinkPayload = null)
    {
        Id = Guid.NewGuid();
        UserId = userId;
        Type = type;
        Title = title;
        Body = body;
        DeepLinkPayload = deepLinkPayload;
        IsRead = false;
        CreatedAt = DateTime.UtcNow;
    }

    public void MarkRead()
    {
        IsRead = true;
    }
}
